### DISCUSSION TITLE
Reconnect Websocket

### DISCUSSION THREAD

**Discussion by abentkamp** [TypeFox/monaco-languageclient#454] [Q&A] [answered]
The blog article https://www.typefox.io/blog/teaching-the-language-server-protocol-to-microsofts-monaco-editor mentions that the `reconnect-websocket` package can be used to automatically reconnect if the connection gets lost. However, the current examples don't show how to do that any more. Any hints on how to do this?

**Comment by kaisalmen** [comment, marked as answer]
Hi @abentkamp thank you for your question. The blog post is from 2017 and the examples and what was written down has diverged. We dropped `reconnect-websocket` to simplify the example and because the lib was no longer maintained. But there are other means to achieve this goal. Using socket.io may be an option: https://socket.io/docs/v4/client-api/

**Reply by abentkamp** [reply]
Ah, ok, good to know. I am guessing that I'll have to implement my own `toSocket` function, but that shouldn't be to hard. Thanks!

**Reply by CGNonofr** [reply]
Note that the LanguageClient class already has a reconnect mechanism, when you create it, you can provide an `errorHandler` in the clientOptions, return ErrorAction.Shutdown in the error method and return CloseAction.Restart in the close method
